PhysicsTools/Heppy/python/physicsobjects/Tau.py

```diff
@@ -44,6 +48,70 @@ def dxy_approx(self, vertex=None):
         return ( - (vtx.x()-vertex.position().x()) *  p4.y()
                  + (vtx.y()-vertex.position().y()) *  p4.x() ) /  p4.pt()
 
+    def countWP(self,name, allWP=False):
+        '''Returns the number of Working Points 
+        that are passed for given isolation name. 
+        '''
+        WPs = ['VVTight', 'VTight', 'Tight', 'Medium',
+               'Loose', 'VLoose', 'VVLoose', ]
+        testname = 'by{}'+name[2:]
+        n_WP = 0
+        for WP in WPs:
+            
+            # if from heppy-implemented isolation (known list of WPs)
+            if allWP:
+                if not self.tauID(testname.format(WP)):
+                    n_WP += 1 # in this case n_WP is the number of not passed WP
+                else:
+                    return len(WPs) - n_WP #stops counting when reached a passed WP
+
+            # else all WPs will be tested from MINIAOD
+            elif self.physObj.isTauIDAvailable(testname.format(WP)):
+                if self.tauID(testname.format(WP)):
+                    n_WP += 1 # in this case n_WP is the number of passed WP
+
+        return n_WP
+
+    def tauID(self, name):
+        '''Returns heppy redefined score if fitting one 
+        of implemented names, else calls the wrapped tau 
+        tauID(name) function.
+        '''
+        if 'IsolationMVArun2017v2DBoldDMwLT' in name:
+            #if raw score
+            if name == 'byIsolationMVArun2017v2DBoldDMwLTraw2017':
+                if self._mvaid_score is None:
+                    self._mvaid_score = self.mva_score()
+                return self._mvaid_score
+
+            #if WPs
+            if self._mvaid_normscore is None:
+                self._mvaid_normscore = self.mva_score(norm=True)
+
```

Comment:
an else should be there. this happens only when not in raw mode and not in auto WP mode